### DISCUSSION TITLE
Consistency in naming: component vs handler

### DISCUSSION THREAD

**Discussion by nickluger** [directus/directus#8210] [Feature Requests]
We just spend half an hour looking for the reason why our custom displays/interfaces didn't get build, until we saw that for custom interfaces, the key in `src/index.js` has to be named  `component`:
```
import InterfaceComponent from './interface.vue';

export default {
	id: 'custom',
	name: 'Custom',
	// => HERE! MUST BE "component"
	component: InterfaceComponent,
	types: ['string'],
};
```

...while for displays it has to be `handler`.

```
import DisplayComponent from './display.vue';

export default {
	id: 'custom',
	name: 'Custom',
        // => HERE! MUST BE "handler"
	handler: DisplayComponent,
	types: ['string'],
};
```

Is there a reason for this inconsistency? Because we used for both cases [Vue 3 Single File Components](https://v3.vuejs.org/guide/single-file-component.html)

**Comment by rijkvanzanten** [comment]
> Is there a reason for this inconsistency? 

The main reason is that the interface is always a Vue component, while the "handler" of a display can also be a function that returns a string to display 🤔

**Reply by rijkvanzanten** [reply]
cc @nickrum, we might want to consolidate that to both be "component" for consistencies sake

**Reply by nickluger** [reply]
Makes kind of sense, but then display should allow `component`, too.

**Reply by rijkvanzanten** [reply]
> Makes kind of sense, but then display should allow component, too.

Yeah, but then we'll have to validate to make sure people don't use both at the same time 🤔 

I think it makes most sense to rename "handler" to "component" for displays, to ensure consistency between the different extension types

**Reply by nickrum** [reply]
I'd be down to consolidate this. But, it might be a little confusing because interfaces can be built with a stateful component or a functional component and displays can be build with a stateful component or a functional handler. So, there is a clear difference between the two.

**Comment by nickrum** [comment]
This has been addressed as part of the Vue 3 migration.
